### PR TITLE
Added a test app to test the Bluetooth LE for Dial from the command line

### PR DESCRIPTION
**Pull request by ajaynavdy**

### CODE REVIEW

**Comment by ajaynavdy** [comment]
@bwill Please review

**Comment by ajaynavdy** [comment]
@pwassam Please review

File: app/src/main/java/com/navdy/hud/bluetoothle/BluetoothLEManager.java [hunk 1]
```diff
@@ -0,0 +1,256 @@
...
+        bluetoothLEScanner = bluetoothAdapter.getBluetoothLeScanner();
+        bluetoothAdapter.getProfileProxy(context, this, HID_PROFILE);
+        registerForBroadcasts();
+    }
+
+    private void registerForBroadcasts() {
+        IntentFilter intent = new IntentFilter();
+        // Bond state
+        intent.addAction(android.bluetooth.BluetoothDevice.ACTION_BOND_STATE_CHANGED);
+        // Bluetooth on and off
```
Comment (bwill) [inline app/src/main/java/com/navdy/hud/bluetoothle/BluetoothLEManager.java]: doesn't this apply to `BluetoothAdapter.ACTION_STATE_CHANGED`?

Comment (ajaynavdy) [inline app/src/main/java/com/navdy/hud/bluetoothle/BluetoothLEManager.java]: You are right. 
intent.addAction(BluetoothAdapter.ACTION_STATE_CHANGED);
intent.addAction(android.bluetooth.BluetoothDevice.ACTION_BOND_STATE_CHANGED);
should be swapped to match the comments

File: app/src/main/java/com/navdy/hud/bluetoothle/BluetoothLEManager.java [hunk 2]
```diff
@@ -0,0 +1,256 @@
...
+        intent.addAction(ACTION_CONNECTION_STATE_CHANGED);
+        // Connectivity
+        intent.addAction(android.bluetooth.BluetoothDevice.ACTION_ACL_CONNECTED);
```
Comment (bwill) [inline app/src/main/java/com/navdy/hud/bluetoothle/BluetoothLEManager.java]: do you need to handle `ACTION_PAIRING_REQUEST`?  We have occasionally gotten a pairing request from the dial bonding process...

File: app/src/main/java/com/navdy/hud/bluetoothle/BluetoothLEManager.java [hunk 3]
```diff
@@ -0,0 +1,256 @@
...
+                Log.e(TAG_BLUETOOTH, "Error ", t);
+            }
+        }
+    };
+
+    public void startScan() {
+        Log.d(TAG_BLUETOOTH, "Starting Scan");
+        scannedBluetoothDevices.clear();
+        scannedDevicesSet.clear();
+        bluetoothLEScanner.flushPendingScanResults(leScanCallback);
```
Comment (bwill) [inline app/src/main/java/com/navdy/hud/bluetoothle/BluetoothLEManager.java]: I would normally expect flush to make sense right before closing, not before starting.

File: app/src/main/java/com/navdy/hud/bluetoothle/BluetoothLEManager.java (hunk 2 above)
Comment (ajaynavdy) [inline app/src/main/java/com/navdy/hud/bluetoothle/BluetoothLEManager.java]: I think we have to handle that case if we get the pairing request, we also need to figure out how to handle that broadcast.

File: app/src/main/java/com/navdy/hud/bluetoothle/BluetoothLEManager.java (hunk 3 above)
Comment (ajaynavdy) [inline app/src/main/java/com/navdy/hud/bluetoothle/BluetoothLEManager.java]: Yeah. Pascal also had the same comment.

File: app/src/main/java/com/navdy/hud/bluetoothle/BluetoothLEManager.java (hunk 2 above)
Comment (bwill) [inline app/src/main/java/com/navdy/hud/bluetoothle/BluetoothLEManager.java]: https://github.com/Navdy/HUD/blob/master/app/src/main/java/com/navdy/hud/app/util/BluetoothUtil.java#L79

Comment (ajaynavdy) [inline app/src/main/java/com/navdy/hud/bluetoothle/BluetoothLEManager.java]: This was very helpful. Thanks.

**Comment by bwill** [comment]
+1

**Pull request merged**